src/cryptography/hazmat/backends/openssl/backend.py

```diff
@@ -1475,7 +1475,9 @@ def _ec_key_set_public_key_affine_coordinates(self, ctx, x, y):
             check_y = self._lib.BN_CTX_get(bn_ctx)
 
             res = set_func(group, point, bn_x, bn_y, bn_ctx)
-            self.openssl_assert(res == 1)
+            if res != 1:
+                self._consume_errors()
+                raise ValueError("EC point not on curve")
```

Comment:
This is a new place where OpenSSL 1.1.0 will reject a key if the point you attempt to set is not on the curve. Previously this occurred in `EC_KEY_check_key` (and still occurs there apparently?), but now it can happen here as well. Rather than giving "not on curve" we could just return "Invalid EC key" though.